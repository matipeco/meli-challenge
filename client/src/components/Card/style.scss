.card{
  &__container{
    background-color: yellow;
  }
}