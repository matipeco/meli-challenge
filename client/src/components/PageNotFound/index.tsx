import { FunctionComponent } from "react"

export const PageNotFound:FunctionComponent = () =>{
  return(
    <h2>Pagina no encontrada</h2>
  )
}