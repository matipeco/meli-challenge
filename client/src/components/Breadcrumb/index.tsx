import { FunctionComponent } from "react";
import './style.scss';

type Props ={
  categories: string[];
}

export const Breadcrumb:FunctionComponent<Props> = ({categories}) => {
  if(categories.length === 0) return null;
  return(
    <ol className="breadcrumb">
     {categories.map((cat)=>{
          return <li key={cat}>
              <a href="#">{cat}</a>
            </li>
        })}
    </ol>
  )
}