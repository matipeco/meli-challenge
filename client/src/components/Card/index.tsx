import { FunctionComponent } from "react"
import { Product } from "../../types"
import './style.scss'

type Props = {
  prod: Product;
}

export const Card:FunctionComponent<Props> = ({prod
})=>{
  return(
    <article className="card__container">
      <h1>{prod.title}</h1>
    </article>
  )
}