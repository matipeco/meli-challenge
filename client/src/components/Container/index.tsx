import { FunctionComponent, PropsWithChildren } from "react"
import "./style.scss"

export const Container:FunctionComponent<PropsWithChildren> = ({children})=>{
  return(
    <div className="container">
      {children}
    </div>
  )
}