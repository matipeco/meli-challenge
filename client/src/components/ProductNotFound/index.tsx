import productNot from '../../assets/productnot.svg'
import './style.scss'

export const ProductNotFound = ()=>{
  return(
      <div className='product-not__div'>
        <article className='product-not__article'>
          <img src={productNot} alt="" width="20" height="20" />
         <div>
            <p>No hay publicaciones que coincidan con tu búsqueda</p>
            <ul>
              <li>Revisá la ortografía de la palabra.</li>
              <li>Utilizá palabras más genéricas o menos palabras.</li>
            </ul>
          </div> 
        </article>
      </div>
  )
}