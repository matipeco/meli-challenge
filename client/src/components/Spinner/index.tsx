import './style.scss';
import spinner from '../../assets/spinner.svg';

export const Spinner = ()=>{
  return(
    <div className='spinner__container'>
      <img src={spinner} alt="cargando" width="50" height="50"/>
    </div>
  )
}