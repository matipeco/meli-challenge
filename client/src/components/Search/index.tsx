import { ChangeEvent, FormEvent, FunctionComponent, useState } from "react"
import "./style.scss"
import lupa from "../../assets/ic_Search.png"
import logo from "../../assets/Logo_ML.png"
import { Container } from "../Container"
import { Link, useNavigate } from 'react-router-dom'

export const Search:FunctionComponent = () =>{
  const navigate = useNavigate();

  const currentPath = window.location.pathname;

  const [input, setInput] = useState<string>("");

  const handleSubmit = (ev:FormEvent<HTMLFormElement>)=>{
    if(input === ""){
      ev.preventDefault();
      return
    }
    ev.preventDefault();
    navigate(`/items?q=${input}`);
  }

  const handleChange = (ev: ChangeEvent<HTMLInputElement>)=>{
    setInput(ev.target.value.toLowerCase())
  }
  return(
      <header className="header">
        <Container>
        {currentPath !=="/" &&
         <Link to={"/"}>
            <img className="header__logo" src={logo} alt="" />
         </Link>
         
         }
        <form action="" className="header__form" onSubmit={handleSubmit}>
          <label htmlFor="input">
            <input id="input" className="header__input" 
            type="text" placeholder="Nunca dejes de buscar" 
            value={input} 
            onChange={handleChange}/>
          </label>
          <button className="header__button">
            <img src={lupa} alt="" className="header__button-logo" />
          </button> 
        </form>
        </Container>
      </header>
  )
} 
