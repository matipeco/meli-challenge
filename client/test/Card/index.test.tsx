import { render } from "@testing-library/react";
import { Card } from "../../src/components/Card";
import { BrowserRouter } from "react-router-dom";

describe("Componente Card", ()=>{
  const product = {
    id: "MLA1408529234",
    title: "Apple iPhone 11 (128 Gb) - Blanco",
    price: {
      currency: "ARS",
      amount: 340000,
      decimals: 0
    },
    picture: "http://http2.mlstatic.com/D_796892-MLA46114829828_052021-I.jpg",
    condition: "new",
    free_shipping: true
  }
  
  test("Se renderiza correctamente", ()=>{
   const { container } = render(
    <BrowserRouter>
      <Card prod={product}/>
    </BrowserRouter>
    )
    expect(container).toMatchSnapshot();
  })
})