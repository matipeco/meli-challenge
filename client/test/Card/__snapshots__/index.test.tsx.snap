// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Card > Se renderiza correctamente 1`] = `
<div>
  <article
    class="card__container"
  >
    <a
      href="/items/MLA1408529234"
    >
      <img
        alt="Apple iPhone 11 (128 Gb) - Blanco"
        class="card__image"
        src="http://http2.mlstatic.com/D_796892-MLA46114829828_052021-I.jpg"
      />
    </a>
    <div
      class="card__div"
    >
      <p
        class="card__contain-price"
      >
        $Â 340.000
        <img
          alt=""
          class="card__contain-shipping"
          src="/src/assets/ic_shipping.png"
        />
      </p>
      <p
        class="card__contain-title"
      >
        Apple iPhone 11 (128 Gb) - Blanco
      </p>
    </div>
    <p
      class="card__location"
    >
      Capital Federal
    </p>
  </article>
</div>
`;
